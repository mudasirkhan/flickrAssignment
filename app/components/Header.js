import styles from "../styles/styles";
import React from "react";
import {View, Text, TextInput, TouchableOpacity, Image, Dimensions } from "react-native";

const {height, width} = Dimensions.get('window');

export default class Header extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            search: props.value,
        }
    }
    render() {
        return (
        <View style={{height: 66, width: width, flexDirection: 'row', alignItems: 'flex-end'}}>
            <View style={styles.header}>
                <TextInput underlineColorAndroid={'transparent'} value={this.state.search} onChangeText={(search) => {
                    this.setState({search})
                }} style={{flex: 1, alignSelf: 'flex-end', marginBottom: 0, paddingHorizontal: 4}}
                           placeholder={"Search"}/>
                <TouchableOpacity onPress={() => {
                    this.props.updateValue(this.state.search)
                }}>
                    <Image source={require("../../assets/search.jpg")} style={styles.image}/>
                </TouchableOpacity>
            </View>
        </View>
        )
    }
}