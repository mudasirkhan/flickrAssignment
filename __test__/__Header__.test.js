// Link.react.test.js
import React from 'react';
import Header from '../app/components/Header';
import renderer from 'react-test-renderer';

function onPress() {
    console.log("wokring")
}
let search = "hello";
let updateValue = (item) => {
    search = item;
}
let searchItem = () => {
    console.log("search")
}
test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Header
            value={search}
            updateValue={updateValue.bind(this)}
            search={searchItem}
            onPress={onPress}
            activeOpacity={1}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    //tree.props.onPress();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});