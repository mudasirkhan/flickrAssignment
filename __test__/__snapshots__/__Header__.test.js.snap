// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "row",
      "height": 66,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#da1e66",
        "borderBottomWidth": 4,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 0,
        "padding": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#da1e66",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingHorizontal": 16,
          }
        }
      >
        Flickr
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Search Images here"
      returnKeyType="go"
      style={
        Object {
          "alignSelf": "flex-end",
          "flex": 1,
          "marginBottom": 0,
          "paddingHorizontal": 4,
        }
      }
      underlineColorAndroid="transparent"
      value="hello"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/close.png",
          }
        }
        style={
          Object {
            "height": 12,
            "width": 12,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/search.png",
          }
        }
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Link changes the class when hovered 2`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "row",
      "height": 66,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#da1e66",
        "borderBottomWidth": 4,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 0,
        "padding": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#da1e66",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingHorizontal": 16,
          }
        }
      >
        Flickr
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Search Images here"
      returnKeyType="go"
      style={
        Object {
          "alignSelf": "flex-end",
          "flex": 1,
          "marginBottom": 0,
          "paddingHorizontal": 4,
        }
      }
      underlineColorAndroid="transparent"
      value="hello"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/close.png",
          }
        }
        style={
          Object {
            "height": 12,
            "width": 12,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/search.png",
          }
        }
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Link changes the class when hovered 3`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "row",
      "height": 66,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#da1e66",
        "borderBottomWidth": 4,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 0,
        "padding": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#da1e66",
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingHorizontal": 16,
          }
        }
      >
        Flickr
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Search Images here"
      returnKeyType="go"
      style={
        Object {
          "alignSelf": "flex-end",
          "flex": 1,
          "marginBottom": 0,
          "paddingHorizontal": 4,
        }
      }
      underlineColorAndroid="transparent"
      value="hello"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/close.png",
          }
        }
        style={
          Object {
            "height": 12,
            "width": 12,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/search.png",
          }
        }
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      />
    </View>
  </View>
</View>
`;
